util/find-doc-nits: Better checking of missing documentation

levitte opened the pull request:
The names collected in util/missing*.txt are not file names, but
symbol names, and to compare properly with script data, the section
name must be included.

All symbols found in util/lib*.num are library functions, so we know
that they are in manual section 3 and can simply add that info.  The
same goes for all macros found in C headers.

Finally, we get rid of getdocced() and its associated hash table
%docced.  We already have the appropriate information in %name_map.

levitte committed:
perl: OpenSSL::Util::Pod::extract_pod_info() now saves the file contents

levitte committed:
util/find-doc-nits: Better checking of missing documentation

The names collected in util/missing*.txt are not file names, but
symbol names, and to compare properly with script data, the section
name must be included.

All symbols found in util/lib*.num are library functions, so we know
that they are in manual section 3 and can simply add that info.  The
same goes for all macros found in C headers.

Finally, we get rid of getdocced() and its associated hash table
%docced.  We already have the appropriate information in %name_map.

levitte committed:
Adjust all util/missing*.txt to include the section number

richsalz commented on util/perl/OpenSSL/Util/Pod.pm:
```diff
@@ -82,6 +82,10 @@ was given as input.
 
 All the names extracted from the NAME section.
 
+=item B<contents =E<gt> "...">
```
oh that's  clever, and avoids double-reading the file.

richsalz commented on util/missingcrypto.txt:
```diff
@@ -1,1423 +1,1423 @@
-v2i_GENERAL_NAME
-v2i_GENERAL_NAMES
-v2i_GENERAL_NAME_ex
+ACCESS_DESCRIPTION_it(3)
```
Since the items in these missingXXXXX files all claim to be *functions* is it really necessary to add the "(3)" after everything?  Can't find-doc-nits append the "(3)" if it needs it internally?  Are you thinking that at some point we'll want "(7)" or such?

levitte replied:
I'm thinking of that possibility, yeah, even if it currently seems remote.

richsalz replied:
So maybe, if/when you want things like "(7)" then make the change at that time?

levitte replied:
And why not now?  I've already made the change, are you actually asking me to undo it?

richsalz replied:
I don't get a vote, but I think it's a needless change.

levitte replied:
I disagree.  It makes things clearer.

levitte committed:
OpenSSL::Util::extract_pod_info(): Read the POD one paragraph at a time

POD files should always be treated this way

levitte committed:
OpenSSL::Util::extract_pod_info(): Allow invisible names

This should be very unusual, but we do have a case of a name we don't
want to display.

levitte committed:
fixup! util/find-doc-nits: Better checking of missing documentation

levitte committed:
doc/man1/openssl-cmds.pod: Add invisble name 'openssl-cmds'

levitte committed:
Rename doc/man7/provider-asymcipher.pod

The correct name is doc/man7/provider-asym_cipher.pod, to match the
name in the NAME section.

levitte commented:
find-doc-nits found a couple of POD errors.  One of them is actually not an error but a conscious choice which needed special attention, while the other just needed a file name tweak.

levitte commented:
Remaining travis failure not relevant (although that particular failure seems to happen consistently, it would be good if someone would look into it)

richsalz commented:
What was the conscious decision (not an error)?

levitte commented:
> What was the conscious decision (not an error)?

To not have `opensl-cmds` in the list of names in `doc/man1/openssl-cmds.pod`

richsalz commented:
> To not have opensl-cmds in the list of names in doc/man1/openssl-cmds.pod

Aha.  I am glad you figured out how to handle that :)

levitte commented:
Ping!

levitte commented:
Ping!

levitte commented:
A rebase on top of 3ce3866389a82837aff169fe3643393367d724a2 might help make this look greener

levitte commented:
(Ping!!!)

levitte commented:
Merged.

5423cabb50edf1ad4c5bba1cbaab8d5fd0826fb7 perl: OpenSSL::Util::Pod::extract_pod_info() now saves the file contents
b4350db5a7ac501e652b75bef21ed3f82d2f6d1b util/find-doc-nits: Better checking of missing documentation
ccfce835e07210064450a58ce189d0e622a1fccf Adjust all util/missing*.txt to include the section number
dfb45dc82479929621c43db921ecfcecc08d3b94 OpenSSL::Util::extract_pod_info(): Read the POD one paragraph at a time
8b849cca31335ae23462beeefe8909b42b168983 OpenSSL::Util::extract_pod_info(): Allow invisible names
b525a7586d933160f252e7a995bd2d08198eccf2 doc/man1/openssl-cmds.pod: Add invisble name 'openssl-cmds'
365955fb27ff7a266d130d145217cdb939b5609a Rename doc/man7/provider-asymcipher.pod
